Release/src/http/client/http_client_curl.cpp

```diff
@@ -0,0 +1,2258 @@
+                }
+                else
+                {
+                    length = std::min(static_cast<size_t>(p_request_context->m_remaining_to_write),
+                                      p_request_context->m_http_client->client_config().chunksize());
+                    if (p_request_context->m_compression_state.m_buffer.capacity() < length)
+                    {
+                        p_request_context->m_compression_state.m_buffer.reserve(length);
+                    }
+                    p_request_context->_get_readbuffer()
+                        .getn(p_request_context->m_compression_state.m_buffer.data(), length)
+                        .then(do_compress)
+                        .then(after_read);
+                }
+            }
+        }
+        else
+        {
+            // We're not compressing; just read and chunk
+            p_request_context->_get_readbuffer()
+                .getn(&p_request_context->m_body_data.get()[http::details::chunked_encoding::data_offset], chunk_size)
+                .then(after_read);
+        }
+    }
+
+    static void _multiple_segment_write_data(_In_ curl_request_context* p_request_context)
+    {
+        auto rbuf = p_request_context->_get_readbuffer();
+        msl::safeint3::SafeInt<utility::size64_t> safeCount = p_request_context->m_remaining_to_write;
+        safeCount = safeCount.Min(p_request_context->m_http_client->client_config().chunksize());
+
+        uint8_t* block = nullptr;
+        size_t length = 0;
+        if (rbuf.acquire(block, length))
+        {
+            if (length == 0)
+            {
+                // Unexpected end-of-stream.
+                if (rbuf.exception() == nullptr)
+                {
+                    p_request_context->report_error(GetLastError(),
+                                                    _XPLATSTR("Error reading outgoing HTTP body from its stream."));
+                }
+                else
+                {
+                    p_request_context->report_exception(rbuf.exception());
+                }
+                return;
+            }
+
+            p_request_context->allocate_request_space(block, length);
+
+            const size_t to_write = safeCount.Min(length);
+
+            // Stop writing chunks after this one if no more data.
+            p_request_context->m_remaining_to_write -= to_write;
+            if (p_request_context->m_remaining_to_write == 0)
+            {
+                p_request_context->m_bodyType = no_body;
+            }
+
+            if (!WinHttpWriteData(p_request_context->m_request_handle,
+                                  p_request_context->m_body_data.get(),
+                                  static_cast<DWORD>(to_write),
+                                  nullptr))
+            {
+                auto errorCode = GetLastError();
+                p_request_context->report_error(errorCode, bui
```

Comment:
@BillyONeal : Can I use ::utility::details::make_unique instead of make_unique from compiler here? This code was copied from http_client_winhttp.cpp

Reply:
seems to work